packages/apps/fortune/exchange-oracle/server/src/modules/job/job.controller.ts

```diff
@@ -74,8 +82,10 @@ export class JobController {
     description: 'Unauthorized. Missing or invalid credentials.',
   })
   @Post('solve')
+  @UseGuards(SignatureAuthGuard)
```

Comment:
You have to specify here which role is to be allowed, e.g. in this case it is worker only.

```suggestion
  @UseGuards(
    new SignatureAuthGuard([Role.Worker]),
  )
```